contributed_definitions/NXreflections.nxdl.xml

```diff
@@ -0,0 +1,415 @@
+      <doc> The experiments from which the reflection data derives </doc>
+    </field>
+
+    <field name="h" type="NX_INT" minOccurs="1">
+      <attribute name="description">
+        <doc> Describes the dataset </doc>
+      </attribute>
+      <doc> The h component of the miller index </doc>
+    </field>
+
+    <field name="k" type="NX_INT" minOccurs="1">
+      <attribute name="description">
+        <doc> Describes the dataset </doc>
+      </attribute>
+      <doc> The k component of the miller index </doc>
+    </field>
+
+    <field name="l" type="NX_INT" minOccurs="1">
+      <attribute name="description">
+        <doc> Describes the dataset </doc>
+      </attribute>
+      <doc> The l component of the miller index </doc>
+    </field>
+
+    <field name="id" type="NX_INT" minOccurs="1">
+      <attribute name="description">
+        <doc> Describes the dataset </doc>
+      </attribute>
+      <doc> 
+        The id of the experiment which resulted in the reflection. If the value
+        is greater than 0, the experiments must link to a multi-experiment NXmx
+        group 
+      </doc>
+    </field>
+
+    <field name="reflection_id" type="NX_INT" minOccurs="1">
+      <attribute name="description">
+        <doc> Describes the dataset </doc>
+      </attribute>
+      <doc>
+        The id of the reflection. Multiple partials from the same reflection
+        should all have the same id
+      </doc>
+    </field>
+    <field name="entering" type="NX_BOOL" minOccurs="1">
+      <attribute name="description">
+        <doc> Describes the dataset </doc>
+      </attribute>
+      <doc> Is the reflection entering or exiting the Ewald sphere </doc>
+    </field>
+    
+    <field name="det_module" type="NX_INT" minOccurs="1">
+      <attribute name="description">
+        <doc> Describes the dataset </doc>
+      </attribute>
+      <doc> The detector module on which the reflection was recorded </doc>
+    </field>
+
+    <field name="flags" type="NX_INT" minOccurs="1">
+      <attribute name="description">
+        <doc> Describes the dataset </doc>
+      </attribute>
+      <doc> Status flags describing the reflection </doc>
+    </field>
+
+    <field name="d" type="NX_FLOAT" minOccurs="1">
+      <attribute name="description">
+        <doc> Describes the dataset </doc>
+      </attribute>
+      <doc> The resolution of the reflection </doc>
+    </field>
+    
+    <field name="partiality" type="NX_FLOAT" minOccurs="1">
+      <attribute name="description">
+        <doc> Describes the dataset </doc>
+      </attribute>
+      <doc> The partiality of the reflection </doc>
+    </field>
+
+    <field name="prd_frame" type="NX_FLOAT" units="frames" minOccurs="1">
+      <attribute name="description">
+        <doc> Describes the dataset </doc>
+      </attribute>
+      <doc> 
+        The frame on which the bragg peak of the reflection is predicted
+      </doc>
+    </field>
+
```

Comment:
repeat this same change on the various ``units`` attributes.  You might suggest in the documentation which units to use in each case.